pkg/controller/command/command_upgrade.go

```diff
@@ -64,40 +84,45 @@ func (r *ReconcileCommand) deleteMigrationJob(commandCR *contrail.Command) error
 	return nil
 }
 
-func getContainerImage(containers []core.Container, name string) string {
-	for _, container := range containers {
-		if name == container.Name {
-			return container.Image
-		}
-	}
-	return ""
-}
-
 func isImageChanged(command *contrail.Command) bool {
 	return command.Status.ContainerImage != "" && command.Status.ContainerImage != getImage(command.Spec.ServiceConfiguration.Containers, "api")
 }
 
-func (r *ReconcileCommand) checkDataMigrationCompleted(command *contrail.Command) (bool, error) {
+func isTargetImageChanged(command *contrail.Command) bool {
+	return command.Status.TargetContainerImage != "" && command.Status.TargetContainerImage != getImage(command.Spec.ServiceConfiguration.Containers, "api")
+}
+
+func (r *ReconcileCommand) checkDataMigrationCompleted(command *contrail.Command) (success bool, failed bool, err error) {
```

Comment:
i would prefer to check if the job finished then the second boolean can say if the result was success or not, so the return values would be (finished, succeeded, err) because success and failure are opposites that can be expressed using one variable and for me it will look better if we first ensure that the job is finished then we can check the result.